### PR TITLE
reports: Add Kriti Rohilla'a GSoC report

### PR DESCRIPTION
**Pull request by kriti21**
https://deploy-preview-612--coala-projects.netlify.com/gsoc/2018/statusreport/kriti21

### CODE REVIEW

File: reports/kriti21.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ [Saurav Singh](https://github.com/damngamerz)
+ [Rafid Aslam](https://github.com/refeed)
+phase:
+ - Bonding : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/9
+ - Phase 1 : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/36
+ - Phase 2 : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/41
+ - Phase 3 : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/42
+bio: >
+ I am in my pre-final year at University Institute of Engineering and Technology,
+ Panjab University. As part of my GSoC 2018 project, I worked on vcs related bears
```
Comment (refeed) [inline reports/kriti21.md]: Capitalize `VCS` as it's an abbreviation

File: reports/kriti21.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ - 5:
+   - repo: coala-bears
+   - link: https://github.com/coala/coala-bears/pull/2543/commits/582e38330d03c696bc7476e102acaa833c9b33af
+   - details: >
+      VCSCommitMetadataBear.py: Add new metabear to analyze git commits.
+ - 6:
+   - repo: coala-bears
+   - link: https://github.com/coala/coala-bears/pull/2576/commits/b51144b2bcc90bbf8a147c6a9e85cc009869bccb
+   - details: >
+      generate_package.py: Replace touch funtion with library function Path.
```
Comment (refeed) [inline reports/kriti21.md]: typo `funtion` -> `function`

File: reports/kriti21.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      CISkipInspectBear.py: Add bear to inspect commits that disable CI builds.
+ - 9:
+   - repo: coala-bears
+   - link: https://github.com/coala/coala-bears/pull/2652
+   - details: >
+      GitRevertInspectBear.py: Add new bear to inspect `git revert` commits.
+---
+
+### Git Commit Content Inspection
+coala has a `GitCommitBear` that is responsible for verifying that commits
```
Comment (refeed) [inline reports/kriti21.md]: `..for verifying commits that..`

File: reports/kriti21.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+`git merge`, `git revert` and commits that disable CI builds. These commits
+are used only under specific conditions and when certain requirements are met.
+With the implementation of these new bears, projects can easily configure
+the `.coafile` to manage such commits according to their requirements.
+
+#### Work Done
+
+<br>
+
+1.) A metadata bear called `VCSCommitMetadataBear` has been implemented which
```
Comment (refeed) [inline reports/kriti21.md]: I think markdown syntax for ordered-list is enough (1. .... (without the `)`)

File: reports/kriti21.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+analyzes the commit at HEAD and returns all relevant information about it
+as `HiddenResult` to the bear using it.
+
+<br>
+
+2.) GitHub PR merge commits were responsible for travis build failure. Settings
```
Comment (refeed) [inline reports/kriti21.md]: `Travis-CI`

File: reports/kriti21.md (hunk 1 above)
Comment (jayvdb) [reply reports/kriti21.md]: And fix indentation (c.f. https://github.com/coala/projects/issues/615)

and add OpenHub social

and start activities at 1 , not 0

**Comment by jayvdb** [comment]
Please finish your PRs

File: _reports/kriti21.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      - details: >
+         CommitBear.py: Fix regex for issue id in CommitBear.
+  - 4:
+      - repo: coala-bears
+      - link: https://github.com/coala/coala-bears/pull/2501/commits/ef0d8e04b55755eccb68219dd61eb28eb0011d83
+      - details: >
+         GitCommitBear.py: Add settings to ignore GitHub PR merge commits.
+  - 5:
+      - repo: projects
+      - link: https://github.com/coala/projects/pull/600
```
Comment (jayvdb) [inline _reports/kriti21.md]: not merged; doesnt count

File: _reports/kriti21.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  - 2018
+  - StatusReport
+report_year: 2018
+date: 2018-08-13
+student: Kriti Rohilla
+organisation: coala
+organisation_link: https://coala.io
+project: Git Commit Content Inspection
+project_link: https://summerofcode.withgoogle.com/projects/#5636421247827968
+tarball: https://raw.githubusercontent.com/kriti21/GSoC-2018-Report/master/ProjectPatches.tar
```
Comment (jayvdb) [inline _reports/kriti21.md]: gzip it pls

File: _reports/kriti21.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+3. `GitLinearCommitBear` has been implemented to detect `git merge` commits
+and inform the author to perform `git rebase` instead in order to keep the
+history clean.
+
+4. Commits that diable CI builds are inspected with `CISkipInspectBear`. Files
+that cannot disable CI build can be controlled by specifying filename patterns
+in `.coafile`.
+
+5. `git revert` commit must be exactly revert of the commit it reverts. For this
```
Comment (refeed) [inline _reports/kriti21.md]: `.. must be exactly the revert ... `

File: _reports/kriti21.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+blog: https://kriti21.github.io/
+activity:
+  - 1:
+      - repo: projects
+      - link: https://github.com/coala/projects/pull/562/commits/52c11a2c85087e705b2a78197aed754d47221174
+      - details: >
+         commit-content-checks.md: Add issue link to project description.
+  - 2:
+      - repo: cEPs
+      - link: https://github.com/coala/cEPs/pull/114/commits/ea25f09ae08b4bc7423453460b14cb16eb3a0db6
```
Comment (jayvdb) [inline _reports/kriti21.md]: this isnt in your tarball.  it should be

and the igitt project .md is missing from this list. it should be included here, as it is hard work, and was related to your current project.  It is also "community work" which is good to include

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/612/commits/5940739b99d45b8a971c11e034d61b5c2d48c328

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**